### PR TITLE
Revert "Make colors generator public to allow for customized colors for chart"

### PR DESCRIPTION
**Pull request by mvorisek**
see #26 discussion

### CODE REVIEW

**Comment by mvorisek** [comment]
@mkrecek234 just to confirm with you, previous PR is useless, right?

**Comment by mkrecek234** [comment]
@mvorisek No, it is not useless. colorGenerator should be public still, only colorGenerator->colors can remain private.

**Comment by mvorisek** [comment]
what is the usecase of public `colorGenerator`?

You can set it using seed like `Chart::addTo($viewOrApp, ['colorGenerator' => new ColorGenerator(), ...])`

**Comment by mkrecek234** [comment]
Simplicity of code - the ColorGenerator class is fine. We just want to be able to adjust color sets dynamically, so access to colorGenerator object should be public. This is just an overkill to me to simply adjust colors. With public colorGenerator we just say:
`$chart->colorGenerator->setAllColorPairs([...])`.

File: src/Chart.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ class Chart extends View
     protected $datasets;
 
     /** @var ColorGenerator */
-    public $colorGenerator;
+    protected $colorGenerator;
```
Comment (mvorisek) [inline src/Chart.php]:
```suggestion
    public $colorGenerator;
```

**Comment by mvorisek** [comment]
Thank you for your feedback, in atk4/ui we do not have strict guidelines for this for now, so we can keep it public as long as there is no getter/setter (and we definitely do not want to extend all code by +100% by adding getter/setter for everything, in short we want getters/setters only for critical parts of code only).

@DarkSide666 https://github.com/atk4/chart/pull/26#issuecomment-1361198024 fixed by this PR, next time please hit the merge button not that fast :) and hope you will recover quickly

**Pull request merged**